lib/deploy/events/apiGateway/restApi.js

```diff
@@ -5,17 +5,55 @@ const BbPromise = require('bluebird');
 
 module.exports = {
   compileRestApi() {
+    if (this.serverless.service.provider.apiGateway &&
+      this.serverless.service.provider.apiGateway.restApiId) {
+      return BbPromise.resolve();
+    }
+
     this.apiGatewayRestApiLogicalId = this.provider.naming.getRestApiLogicalId();
 
+    let endpointType = 'EDGE';
+
+    if (this.serverless.service.provider.endpointType) {
+      const validEndpointTypes = ['REGIONAL', 'EDGE', 'PRIVATE'];
```

Comment:
Selecting endpointType is great!
However, those are not for authorizer feature.
Could you separate PR for each feature?